fg/apps/main/templates/maps/orders.html

```diff
@@ -1,64 +1,76 @@
 {% extends "base/page.html" %}
 {% load staticfiles %f}
+
 {% block content %}
+  {% include "messages/message.html" %}
+  <div class="container" style="padding-top: 100px;">
+    <h1>Orders Map</h1>
+    <div id='map' style='width: 800px; height: 600px;'></div>
+  </div>
+{% endblock %}
 
+{% block pagescripts %}
 <style>
-.node rect {
-  cursor: pointer;
-  fill: #fff;
-  fill-opacity: 0.5;
-  stroke: #3182bd;
-  stroke-width: 1.5px;
-}
-
-.node text {
-  font: 10px sans-serif;
-  pointer-events: none;
-}
+  .marker {
+    background-color: #FA4653;
+    background-size: cover;
+    border-radius: 50%;
+    cursor: pointer;
+  }
+  .mapboxgl-popup {
+    max-width: 200px;
+  }
 
-.link {
-  fill: none;
-  stroke: #9ecae1;
-  stroke-width: 1.5px;
-}
+  .mapboxgl-popup-content {
+    text-align: center;
+    font-family: 'Open Sans', sans-serif;
+  }
 </style>
+<script src='https://api.tiles.mapbox.com/mapbox-gl-js/v1.3.1/mapbox-gl.js'></script>
+<link href='https://api.tiles.mapbox.com/mapbox-gl-js/v1.3.1/mapbox-gl.css' rel='stylesheet' />
 
-{% include "messages/message.html" %}
-<div class="container">
-   <h1>Orders Map</h1>
-   <hr>
-   <div id="map" style="position: relative; width: 800px; height: 600px;"></div>
-</div>
-{% endblock %}
-{% block pagescripts %}
-<script src="//cdnjs.cloudflare.com/ajax/libs/d3/3.5.3/d3.min.js"></script>
-<script src="//cdnjs.cloudflare.com/ajax/libs/topojson/1.6.9/topojson.min.js"></script>
-<script src="//cdnjs.cloudflare.com/ajax/libs/datamaps/0.5.9/datamaps.world.min.js"></script>
 <script>
+  mapboxgl.accessToken = '{{ mapbox_token }}';
+
+  var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/light-v10',
+    center: [-95, 40],
+    zoom: 3,
+  });
+
+  var geojson = {
+    type: 'FeatureCollection',
+    features: [
+    {% for zip,datum in data.items %}
+      {
+        type: 'Feature',
+        geometry: {
+          type: 'Point',
+          coordinates: [{{ datum.long }}, {{ datum.lat }}]
+        },
+        properties: {
+          title: 'OpenEnzyme',
+          description: 'A test shipment of OpenEnzyme to a community lab.',
+        },
+        size: {
+          width: Math.sqrt({{datum.count}})*15,
+          height: Math.sqrt({{datum.count}})*15,
+        }
+      },
+    {% endfor %}
```

Comment:
This endfor should be moved up so it's directly after the comma.

Also - you should be checking for `{% if forloop.last %}` to *not* render the comma.

Reply:
> This endfor should be moved up so it's directly after the comma.

Will make this change as mentioned above.

> Also - you should be checking for {% if forloop.last %} to not render the comma.

I chose not to do this because it overcomplicates the code and because JavaScript objects both allow and encourage trailing commas for simplicity and in order to make it harder to make editing errors. This is in contrast to the JSON specification which prohibits trailing commas.

Reply:
Oh that's really interesting - I wasn't aware of this! Yes I'm totally okay with that.